rosbag2_compression/src/rosbag2_compression/zstd_decompressor.cpp

```diff
@@ -175,6 +176,35 @@ void throw_on_zstd_error(const ZstdDecompressReturnType compression_result)
   }
 }
 
+/**
+ * Checks rcutils array resizing and throws a runtime_error if there was an error resizing.
+ * \param rcutils_ret_t Result of calling rcutils
+ */
+void throw_on_rcutils_resize_error(const rcutils_ret_t resize_result)
```

Comment:
Can we put this in a common file instead of duplicating it? Maybe some something like `compression_utils.hpp`

Reply:
There's actually quite a few duplicated helper functions in here and `zstd_compressor.cpp`; I don't mind consolidating them all, but @emersonknapp , would you rather I do it in this PR or in a new one?

Reply:
If you plan on consolidating them all and have a follow up PR then that is better.
Feel free to open a draft stacked PR if you already have the changes.

Reply:
https://github.com/ros2/rosbag2/pull/459